File: replace_string.py
import sys
import os

def replace_in_file(path, old, new):
    if not os.path.exists(path):
        print(f"File not found: {path}")
        return
    try:
        with open(path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if old in content:
            new_content = content.replace(old, new)
            with open(path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated {path}")
        else:
            print(f"String '{old}' not found in {path}")
    except Exception as e:
        print(f"Error processing {path}: {e}")

if __name__ == "__main__":
    if len(sys.argv) < 4:
        print("Usage: python replace.py <file> <old> <new>")
        sys.exit(1)
    
    path = sys.argv[1]
    old = sys.argv[2]
    new = sys.argv[3]
    replace_in_file(path, old, new)

File: replace_string.js
const fs = require('fs');
const path = process.argv[2];
const oldStr = process.argv[3];
const newStr = process.argv[4];

if (!path || !oldStr || !newStr) {
    console.log("Usage: node replace.js <file> <old> <new>");
    process.exit(1);
}

if (!fs.existsSync(path)) {
    console.log(`File not found: ${path}`);
    // Don't exit with error if file not found, just skip (for qa-ci.yml if it doesn't exist)
    process.exit(0);
}

try {
    let content = fs.readFileSync(path, 'utf8');
    if (content.includes(oldStr)) {
        content = content.split(oldStr).join(newStr);
        fs.writeFileSync(path, content, 'utf8');
        console.log(`Updated ${path}`);
    } else {
        console.log(`String '${oldStr}' not found in ${path}`);
    }
} catch (e) {
    console.error(`Error processing ${path}: ${e}`);
    process.exit(1);
}
